docs/integration/solrcloud.md

```diff
@@ -1,9 +1,112 @@
+
+The Sematext SolrCloud Monitoring Agent can be run in two different modes - *in-process* and *standalone*. The *in-process* one is run as a Java agent, it is simpler to initially setup, but will require restring your Solr node when you will want to upgrade your monitoring Agent, i.e. to get new features. The benefit of *standalone* agent mode is that it is running as a separate process and doesn't require Solr restart when it is upgraded.
+
+After creating a [SolrCloud App in Sematext](https://apps.sematext.com/ui/monitoring-create) you need to install the Monitoring Agent on each host that is running your SolrCloud nodes to have the full visibility over the metrics from each host. The full installation instructions can be found in the [setup instructions](https://apps.sematext.com/ui/howto/SolrCloud/overview) displayed in the UI. 
+
+For example, on CentOS, you need to add Sematext Linux packages and install them with the following command:
+```bash
+sudo wget https://pub-repo.sematext.com/centos/sematext.repo -O /etc/yum.repos.d/sematext.repo
+sudo yum clean all
+sudo yum install spm-client
+``` 
+
+After that you need to setup the SolrCloud Monitoring Agent by running a command like this:
+```bash
+sudo bash /opt/spm/bin/setup-sematext  \
+    --monitoring-token <your-monitoring-token-goes-here>   \
+    --app-type solrcloud  \
+    --agent-type javaagent  \
+    --infra-token <your-infra-token-goes-here>
+```
+
+The above command will setup your SolrCloud Monitoring Agent in the *in-process* mode, to have it running in the *standalone* mode instead of running the above command, run the one shown below:
+```bash
+sudo bash /opt/spm/bin/setup-sematext  \
+    --monitoring-token <your-monitoring-token-goes-here>   \
+    --app-type solr  \
+    --agent-type standalone  \
+    --infra-token <your-infra-token-goes-here>  \
+    --jmx-params '-Dspm.remote.jmx.url=localhost:3000'
+```
+
+Keep in mind that your need to provide the Monitoring token and Infra token. They are both provided in the [installation instructions](https://apps.sematext.com/ui/howto/SolrCloud/overview) for your SolrCloud App. 
+
+Finally, the last thing that needs to be done is adjusting the *solr.in.sh* file and add the following section:
+```bash
+SOLR_OPTS="$SOLR_OPTS -Dcom.sun.management.jmxremote -javaagent:/opt/spm/spm-monitor/lib/spm-monitor-generic.jar=<your-monitoring-token-goes-here>::default"
+```
+
+Or if you would like to run the Solr Monitoring Agent in the *standalone* mode add the following section to the ```solr.in.sh``` file:
+```bash
+SOLR_OPTS="$SOLR_OPTS -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=3000 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false"
+```
+
+Make sure that tag ```<jmx />``` is enabled in your ```solrconfig.xml``` file.
+
+**You need to restart your SolrCloud node after the above changes.**
```

Comment:
**You need to restart your SolrCloud node after the _changes above_ * .**

Reply:
Adjusted.